R/mapping_using_bigQuery_drugExposure.R

```diff
@@ -0,0 +1,116 @@
+# Load necessary libraries
+library(dplyr)
+library(DBI)
+library(readr)
+library(AMR)  # For drug name mapping
+
+# Load data from CSV files
+drug_exposure <- read_csv("../data/OMOP/cleaned_drug_exposure.csv")
+concept <- read_csv("../data/OMOP/cleaned_concept.csv")
+
+
+
+# Print the column names to check if they are correct
+print(colnames(concept))
+# Print the cleaned data to verify everything looks correct
+print(head(concept))
+
+
+# Ensure the relevant columns are available in drug_exposure
+print(colnames(drug_exposure))
+print(head(drug_exposure))  
+
+
+#drug_exposure <- drug_exposure %>%
+#  left_join(concept, by = c("drug_concept_id" = "concept_id")) %>%
+#  rename(drug_name = concept_name) %>%
+#  select(-domain_id, -vocabulary_id, -concept_class_id, -standard_concept, -concept_code, 
+#         -valid_start_date, -valid_end_date, -invalid_reason)
+
+#Mapping using left join
+drug_exposure <- drug_exposure %>%
+  left_join(concept, by = c("drug_concept_id" = "concept_id")) %>%
+  rename(drug_name = concept_name) %>%  # Rename concept_name to drug_name
+  mutate(route = NA)  # Since route_concept_id is NA.

```

Comment:
```suggestion
#This block joins the drug_exposure table with the concept table based on matching drug_concept_id and #concept_id. After joining, it renames the concept_name column to drug_name for clarity and initialises a route #column with NA since the route_concept_id is missing in this dataset.
drug_exposure <- drug_exposure %>%
  left_join(concept, by = c("drug_concept_id" = "concept_id")) %>%
  rename(drug_name = concept_name) %>%  # Rename concept_name to drug_name
  mutate(route = NA)  # Since route_concept_id is NA.

```